src/resolve/index.ts

```diff
@@ -6,17 +6,73 @@ import {Got} from '../network/got'
 import {Package} from '../types'
 import {LoggedPkg} from 'pnpm-logger'
 
-export type ResolveResult = {
+export type ResolutionBase = {
   id: string,
+}
+
+/**
+ * npm registry hosted package
+ */
+export type PackageResolution = ResolutionBase & {
+  type: 'package',
+  tarball: string,
+  shasum?: string,
   pkg?: Package,
+}
```

Comment:
yes, probably. This would also solve the other issue that I commented under shrinkwrap